File: CSharpCodeBase/components/controlsystem/enemyactions/actionbehaviour.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class actionbehaviour :action{
 public void init (self, entity, tree);
       Action.init(self, 0, true, "behaviour");
       this.entity = entity;
       this.tree = tree;
 }})
 public void Update(){
   this.tree:Process();
   return false;
 }
 public void OnStopRunning(){
   this.tree:Reset();
 }
 public void OnEvent(data){
   this.tree:OnEvent(data);
 }
 }}

File: CSharpCodeBase/components/controlsystem/enemyactions/actionbehaviour.lua
local Action = require 'components.controlsystem.action'
local Class = require 'utils.hump.class'

local ActionBehaviour = Class({__includes=Action, init = function (self, entity, tree)
      Action.init(self, 0, true, 'behaviour')
      self.entity = entity
      self.tree = tree
end})

function ActionBehaviour:Update()
  self.tree:Process()
  return false
end

function ActionBehaviour:OnStopRunning()
  self.tree:Reset()
end

function ActionBehaviour:OnEvent(data)
  self.tree:OnEvent(data)
end


return ActionBehaviour